addons/website/static/src/scss/website.scss

```diff
@@ -2446,3 +2446,43 @@ input[value*="data-oe-translation-initial-sha"] {
 .o_dropdown_without_offset {
     transform: none !important;
 }
+
+// Subtle input-group-text
+// This can be set to an input-group-text placed directly before or after the input.
+.input-group-text.input-group-text-subtle {
+    @include transition($input-transition);
+    border-color: $input-border-color;
+    background: $input-bg;
+    color: $input-color;
+
+    > * {
+        color: rgba($input-color, 0.6);
+    }
+
+    .is-invalid ~  &,
+    &:has(~ .is-invalid) {
```

Comment:
`:has` should not be used:
- It rarely make sense for CSS IMO.
- It does not work on my Firefox even though it supposed to be auto-updated. And other browsers (~3% of website users will have this not working). Unless you have no other alternative, don't use it. If you use it, make sure the result is not awful without it :wink: 

In this case, I don't understand the rule "input-group-text which has a child who has a neightbor on its left which is invalid"... is that even possible? :thinking: Something invalid inside the input-group-text itself?

Reply:
> In this case, I don't understand the rule "input-group-text which has a child who has a neightbor on its left which is invalid"... is that even possible?

We think the best approach would be to add a custom class on the `input-group-text` when we need something more subtle.
This is the cleanest solution we've found for selecting the element **preceding** the `form-control` .
Here, the selector targets the `input-group-text` preceding the `form-control.is-invalid`.
<img width="659" alt="Capture d’écran 2024-02-13 à 16 49 40" src="https://github.com/odoo/odoo/assets/80679690/af6d8c26-b57e-4441-a57b-510ec07efd4e">

> Unless you have no other alternative, don't use it.

Would you have a better idea ? :)

> If you use it, make sure the result is not awful without it

Okay, I'll check the browsers that don't support it 👍

Reply:
It's a pity they don't give this possibility, it would be really nice 😐.

But stylizing the invalid state is necessary for the `input-group-text-subtle`, since they "merge" with the `form-control` 🙂 .

Reply:
I guess it's ok if this does not work on all Firefox, let's keep it.